spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+    String backupName =
+        String.format(
+            "%s_%02d",
+            sourceDatabaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(sourceDatabaseId.getInstanceId(), backupName);
+    Backup backup = dbAdminClient.newBackupBuilder(backupId).build();
+    // Initiate the request which returns an OperationFuture.
+    System.out.println(String.format(
+        "Restoring backup [%s] to database [%s]...",
+        backup.getId().toString(),
+        restoreToDatabase.toString()));
+    // Try the restore operation in a retry loop, as there is a limit on the number of restore
+    // operations that is allowed to execute simultaneously, and we should retry if we hit this,
+    // limit.
+    int restoreAttempts = 0;
+    while (true) {
+      try {
+        OperationFuture<Database, RestoreDatabaseMetadata> op = backup.restore(restoreToDatabase);
+        // Wait until the database has been restored.
+        Database db = op.get();
+        // Refresh database metadata and get the restore info.
+        RestoreInfo restore = db.reload().getRestoreInfo();
+        System.out.println(
+            "Restored database ["
+                + restore.getSourceDatabase().getName()
+                + "] from ["
+                + restore.getBackup().getName()
+                + "]");
+        break;
+      } catch (ExecutionException e) {
+        SpannerException se = (SpannerException) e.getCause();
+        if (se.getErrorCode() == ErrorCode.FAILED_PRECONDITION
+            && se.getMessage()
+                .contains("Please retry the operation once the pending restores complete")) {
+          restoreAttempts++;
+          if (restoreAttempts == 10) {
+            System.out.println(
+                "Restore operation failed 10 times because of other pending restores. "
+                + "Giving up restore.");
+            return;
+          }
+          // Wait and then retry.
+          System.out.println(String.format(
+              "Restoring backup %s to database %s must wait because of other pending restore",
+              backup.getId().getBackup(),
+              restoreToDatabase.getDatabase()));
+          Uninterruptibles.sleepUninterruptibly(60L, TimeUnit.SECONDS);
+        } else {
+          throw se;
+        }
+      } catch (InterruptedException e) {
+        throw SpannerExceptionFactory.propagateInterrupt(e);
+      }
+    }
+  }
+  // [END spanner_restore_backup]
+
+  // [START spanner_update_backup]
+  static void updateBackup(DatabaseAdminClient dbAdminClient, DatabaseId databaseId) {
+    // Get the most recent sample backup of the sample database.
+    String backupName =
+        String.format(
+            "%s_%02d",
+            databaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(databaseId.getInstanceId(), backupName);
```

Comment:
Same here, it would be good to get the backupId as an input.

Reply:
done.